environment/src/main/java/com/sequenceiq/environment/environment/flow/deletion/handler/EnvironmentUMSResourceDeleteHandler.java

```diff
@@ -0,0 +1,59 @@
+package com.sequenceiq.environment.environment.flow.deletion.handler;
+
+import static com.sequenceiq.environment.environment.flow.deletion.event.EnvDeleteHandlerSelectors.DELETE_UMS_RESOURCE_EVENT;
+import static com.sequenceiq.environment.environment.flow.deletion.event.EnvDeleteStateSelectors.FINISH_ENV_DELETE_EVENT;
+
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.altus.GrpcUmsClient;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.environment.flow.deletion.event.EnvDeleteEvent;
+import com.sequenceiq.flow.reactor.api.event.EventSender;
+import com.sequenceiq.flow.reactor.api.handler.EventSenderAwareHandler;
+
+import reactor.bus.Event;
+
+@Component
+public class EnvironmentUMSResourceDeleteHandler extends EventSenderAwareHandler<EnvironmentDto> {
```

Comment:
Can you please cover it with unit tests?

Reply:
ok, in progress..

Reply:
done